packages/eslint-plugin/rules/jsdoc-tag-order.test.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * External dependencies
+ */
+import { RuleTester } from 'eslint';
+
+/**
+ * Internal dependencies
+ */
+import rule from './jsdoc-tag-order';
+
+const ruleTester = new RuleTester( {
+	parserOptions: {
+		sourceType: 'module',
+		ecmaVersion: 6,
+	},
+} );
+
+ruleTester.run( 'jsdoc-tag-order', rule, {
+	valid: [
+		{
+			code: `
+/**
+ * Wraps children components with a fresh test registry,
+ * which can be configured by its callback prop.
+ *
+ * @since 1.7.1
+ * @private
+ *
+ * @param {?Object}   props          Component props.
+ * @param {?Function} props.callback Function which receives the registry instance.
+ * @param {?Object}   props.registry Registry object; uses \`createTestRegistry()\` by default.
+ * @return {string} A test string.
+ */
+export function WithTestRegistry( { children, callback, registry = createTestRegistry() } = {} ) {
+	// Populate most basic data which should not affect any tests.
+	provideUserInfo( registry );
+
+	if ( callback ) {
+		callback( registry );
+	}
+
+	return 'test';
+}
```

Comment:
Again, this seems much more verbose than necessary and mostly adds noise to the test.
Does ESLint complain when things aren't used here as well? Is it checking the given code with all of our rules or just the one? It's at least using undefined functions so we could always do something like `dummyFunction( paramA, paramB, paramC )` if needed.